HeatStroke/Graphics/Renderable/TextBox.cpp

```diff
@@ -0,0 +1,389 @@
+	{
+		if (p_fWidth != m_fWidth || p_fHeight != m_fHeight)
+		{
+			m_fWidth = p_fWidth;
+			m_fHeight = p_fHeight;
+			CreateQuads();
+		}
+	}
+
+	//----------------------------------------------------------
+	// Sets alignment for text (top left by default)
+	//----------------------------------------------------------
+	void TextBox::SetAlignment(const HORIZONTAL_ALIGNMENT& p_eHorizontal, const VERTICAL_ALIGNMENT& p_eVertical)
+	{
+		if (p_eHorizontal != m_eHorizontal || p_eVertical != m_eVertical)
+		{
+			m_eHorizontal = p_eHorizontal;
+			m_eVertical = p_eVertical;
+			CreateQuads();
+		}
+	}
+
+	//----------------------------------------------------------
+	// Sets string for textbox
+	//----------------------------------------------------------
+	void TextBox::SetText(const std::string& p_strMessage)
+	{
+		//TODO: printf like functionality
+		if (p_strMessage != m_strMessage)
+		{
+			m_strMessage = p_strMessage;
+			CreateQuads();
+		}
+	}
+
+	//----------------------------------------------------------
+	// Sets position for textbox (origin at center), measured from top-left of screen
+	//----------------------------------------------------------
+	void TextBox::SetPosition(const GLfloat& p_fXPos, const GLfloat& p_fYPos)
+	{
+		m_vPosition = glm::vec2(p_fXPos, 720.0f-p_fYPos);
+		glm::mat4 mWorld = glm::translate(glm::vec3(m_vPosition, 0.0f));
+
+		//iterate through all renderable quads and set the matrix uniform
+		for (std::vector<Texture_Page_Quads>::iterator iter = m_mQuads.begin(); iter != m_mQuads.end(); ++iter)
+		{
+			iter->m_mWorld = mWorld;
+		}
+	}
+
+	//----------------------------------------------------------
+	// Sets colour for string in textbox
+	//----------------------------------------------------------
+	void TextBox::SetColour(const glm::vec4& p_vColour)
+	{
+		m_vColour = p_vColour;
+
+		//iterate through all renderable quads and set the color uniform
+		for (std::vector<Texture_Page_Quads>::iterator iter = m_mQuads.begin(); iter != m_mQuads.end(); ++iter)
+		{
+			iter->m_pMaterial->SetUniform("UnifColor", m_vColour);
+		}
+	}
+
+	//----------------------------------------------------------
+	// 'Wrapper' for creating quads to display text, filling up m_Quads with data to render
+	//----------------------------------------------------------
+	void TextBox::CreateQuads()
+	{
+		static const GLfloat s_fTEXT_SCALING_SHRINK = 0.05f;	//how much to shrink the text scale per shrinkage
+		static const GLfloat s_fTEXT_SCALING_MIN = 0.1f;		//the smallest you can shrink the text
+
+		//obviously can't create the quads
+		if (m_fHeight <= 0.0f || m_fWidth <= 0.0f)
+		{
+			assert(false && "Height or Width must be greater than 0 for textbox.");
+			return;
+		}
+		
+		//---------------------------
+		// Separate text into lines and create source information for Quads
+		//---------------------------
+
+		std::vector<std::string> mLines;
+		std::vector<glm::vec2> mLineSizes;
```

Comment:
The answer to both your questions is because I did them last year, so they followed my own naming convention rather than the one we decided this year. I uploaded it with minimum changes (except to comply to the HeatStroke namespace) and will probably leave it as is rather than revisit it just to rename variables.